internal/testsupport/files.go

```diff
@@ -0,0 +1,40 @@
+package testsupport
+
+import (
+	"os"
+	"testing"
+)
+
+func Setfiles(t *testing.T, files map[string]string) func() {
```

Comment:
```suggestion
// SetFiles can be used to temporarily create files on the local file system.
// It returns a function that will clean up all files it created.
func SetFiles(t *testing.T, files map[string]string) func() {
```

What do you think about creating the files in `/tmp` (not sure about Windows :thinking:), to make sure that they get cleaned up one way or another?